src/platform.ts

```diff
@@ -101,52 +117,32 @@ export class MiHomeGateway implements DynamicPlatformPlugin {
           // the `context` property can be used to store any data about the accessory you may need
           accessory.context.device = device;
 
-          const platformAccessory = this.createAccessoryHandler(accessory);
+          // create the accessory handler for the newly create accessory
+          this.newPlatformAccessory(accessory);
 
-          if (platformAccessory !== null) {
-            // link the accessory to your platform
-            this.api.registerPlatformAccessories(PLUGIN_NAME, PLATFORM_NAME, [accessory]);
-
-            // push into accessory cache
-            this.accessories.push(accessory);
-          }
+          // link the accessory to your platform
+          this.api.registerPlatformAccessories(PLUGIN_NAME, PLATFORM_NAME, [accessory]);
         }
       }
     } catch (err) {
       this.log.debug('Error discovering devices', err);
     }
   }
 
-  private createAccessoryHandler(accessory: PlatformAccessory): MiHomePlatformAccessory | null {
-    const device: SubDevice = accessory.context.device;
-
-    if (this.isNullOrEmpty(device)) {
-      this.log.warn('Device context is null, unable to create accessory');
-      return null;
-    }
-
-    try {
-      switch (device.device_type) {
-        case DeviceType.CONTROL:
-        case DeviceType.LEGACY:
-        case DeviceType.LIGHT:
-        case DeviceType.RELAY:
-          return new SwitchAccessory(this, accessory);
-        default:
-          this.log.warn('Accessory type not supported: %s [%s]', device.label, device.device_type);
-          return null;
-      }
-    } catch (err) {
-      this.log.error('Error creating accessory handler', err);
-      return null;
-    }
-  }
-
   /**
-   * Check if object is null, empty, or undefined.
-   * @param obj Object.
+   * create the accessory handler for the newly create accessory
+   * @param accessory Platform accessory.
    */
-  private isNullOrEmpty<T>(obj: T): boolean {
-    return obj === null || obj === undefined || Object.keys(obj).length === 0;
+  protected newPlatformAccessory(accessory: PlatformAccessory) {
+    switch (accessory.context.device_type) {
```

Comment:
Should be `accessory.context.device.device_type`

Reply:
Thanks for spotting this